Remove post-completion result collection, add autostart option

pmoeller91-yottaa opened the pull request:
feat!: adds ability to autostart QUnit tests

BREAKING CHANGE: To avoid losing test context, removes the ability to collect test results if the tests are already completed in the browser when the service is inserted. Instead, a recommendation is made to disable test auto-start in QUnit and allow the service to start the tests itself. If this is infeasible, adding a manual delay in the tests will also serve to give the service time to enter the browser context prior to test completion.

# Summary

Prior, when test results were collected post-completion, this resulted in a different format for the result which lost module names, and some information missing from the results. My answer to this is to encourage delayed test start by disabling `autostart` in QUnit and instead allowing the `wdio-qunit-service` to handle automatically starting the test. This will guarantee consistent results from QUnit.

In the case where tests must be started without using the service a couple issues may result.

Issue: Tests start and finish before the test service can setup the callback

Possible fixes:
 - Modify tests so they can be started by the QUnit WDIO service
 - Add a slight manual delay to the beginning of the test run to ensure the service can attach the callback prior to test run completion.
   - Ex: `await new Promise((resolve) => { setTimeout(resolve, 250); });` in a `before` hook.

Issue: Tests are started prior to being ready by the test service

Possible fixes:
 - Extend the `autostartDelay` value to allow additional time for the tests to automatically be started in the browser by your mechanism of choice. The service will only attempt to autostart if the tests have not started at the end of this delay.

```js
// in wdio.conf.js
export const config = {
  // ..
  services: [['qunit', { autostartDelay: 1000 }],
  // ..
};
```

pmoeller91-yottaa committed:
feat!: adds ability to autostart QUnit tests

BREAKING CHANGE: To avoid losing test context, removes the ability
to collect test results if the tests are already completed in the
browser when the service is inserted. Instead, a recommendation is
made to disable test auto-start in QUnit and allow the service to
start the tests itself. If this is infeasible, adding a manual delay
in the tests will also serve to give the service time to enter the
browser context prior to test completion.

pmoeller91-yottaa committed:
docs: amend README with fixed style and information

pmoeller91-yottaa committed:
chore: update workflows and docs to use new package name

Updates workflows and modifies documentation to reflect the
decision to publish the package to the @yottaa org using
GitHub packages.

pmoeller91-yottaa committed:
chore: remove CODEOWNERS file

Generally not used in our org and no appropriate team or individual is available.

bburke4 commented on .github/workflows/test.yml:
```diff
@@ -22,14 +23,28 @@ jobs:
+        run: npm ci --ignore-scripts
+      - name: Build
+        run: npm run build
+      - name: "Test: ESM"
```
the names are optional... it just uses the command in the action output/logs, but doesnt matter

pmoeller91-yottaa replied:
Yeah just makes it look a little nicer, if you have objections let me know.

bburke4 commented on package.json:
```diff
@@ -67,9 +69,12 @@
   },
   "bugs": {
-    "url": "https://github.com/mauriciolauffer/wdio-qunit-service/issues"
+    "url": "https://github.com/pmoeller91-yottaa/wdio-qunit-service/issues"
```
should change to yottaa? or delete i guess

pmoeller91-yottaa replied:
Fixed!

bburke4 commented on README.md:
```diff
@@ -1,8 +1,11 @@
 
-[![npm](https://img.shields.io/npm/v/wdio-qunit-service)](https://www.npmjs.com/package/wdio-qunit-service) [![test](https://github.com/mauriciolauffer/wdio-qunit-service/actions/workflows/test.yml/badge.svg)](https://github.com/mauriciolauffer/wdio-qunit-service/actions/workflows/test.yml)
+[![NPM Version](https://img.shields.io/npm/v/@yottaa/wdio-qunit-service)](https://www.npmjs.com/package/@yottaa/wdio-qunit-service)
+[![test](https://github.com/pmoeller91-yottaa/wdio-qunit-service/actions/workflows/test.yml/badge.svg)](https://github.com/pmoeller91-yottaa/wdio-qunit-service/actions/workflows/test.yml)
```
change org to yottaa?

pmoeller91-yottaa replied:
Good catch, fixed this and a couple other things I missed. See latest commit. Had to force push to fix some commit messages.

pmoeller91-yottaa committed:
build: add commitlint, update docs to point to Yotta instead of pmoeller91-yottaa

This change adds commitlint to best support the release-it action, by adding a commit-msg hook
locally as well as a check as part of the test step in the Test GitHub action. Documentation is
updated to point universally to Yottaa instead of specifically to pmoeller91-yottaa to reflect the
package ownership transfer.
